compiler/dalgona/include/PreOperatorHook.h

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __DALGONA_PRE_OPERATOR_HOOK_H__
+#define __DALGONA_PRE_OPERATOR_HOOK_H__
+
+#include "Utils.h"
+
+#include <luci_interpreter/Interpreter.h>
+#include <luci_interpreter/core/Tensor.h>
+#include <luci/IR/CircleNodeVisitor.h>
+#include <luci/IR/CircleNodeDecl.h>
+
+#include <pybind11/embed.h>
+#include <pybind11/numpy.h>
+#include <vector>
+#include <unordered_map>
+
+using Tensor = luci_interpreter::Tensor;
+
+namespace py = pybind11;
+using namespace py::literals;
+
+namespace dalgona
+{
+class PreOperatorHook final : public luci::CircleNodeVisitor<void>
+{
+
+// This macro describes common tasks inside pre-opeartor hook
+// After this macro, two variables are created
+// 1. hook: Python function called before operator execution (type: py::object)
+// 2. inputs: input data (type: std::vector of numpy array)
+#define PRE_OPERATOR_HOOK_PROLOGUE(OP_NAME)             \
+  if (!py::hasattr(_analysis, #OP_NAME "Pre"))          \
+  {                                                     \
+    visit(static_cast<const luci::CircleNode *>(node)); \
+    return;                                             \
+  }                                                     \
+  py::object hook = _analysis.attr(#OP_NAME "Pre");     \
+  auto inputs = inputsPyArray(node, _interpreter);
+
+private:
+  py::object _analysis;
+  luci_interpreter::Interpreter *_interpreter;
```

Comment:
initialize variable

Reply:
These variables are initialized when `PreOperatorHook` is constructed.

Reply:
OK, please check with our static code analysis, SVACE.